soluce_mentorat.py

```diff
@@ -26,20 +26,36 @@ def is_taken(self, x: int, y: int) -> bool:
         if status_position == Status.EMPTY or status_position == Status.OUT:
             return False
 
+        # Initialisation de l'historique des positions testées et de l'état de la forme
+        if positions is None:
+            positions = []
+            self.shape_is_free = False
```

Comment:
Une bonne pratique en Python est de déclarer les membres de classes (en préalable d'utilisation) dans `__init__(...)`
![image](https://user-images.githubusercontent.com/7001058/177137705-cead812b-b975-4e6f-8f68-c3e3d7c0c48e.png)

Reply:
Une autre bonne pratique est de considérer le scope d'utilisation de tes membres de classe.
Par exemple, `self.shape_is_free` semble être un outil en interne pour tracker le status d'une forme en cours d'évaluation. 
Ça ressemble énormément à un outil interne, quelque chose que l'utilisateur de ta classe ne doit pas utilisé (diretement) ou altéré. 
En python, il n'y a pas (vraiment) de notion de protected ou private méthodes/membres/... 
Mais par convention on peut spécifier avec des `_` le statut d'une fonction/méthode/membre de classe => [doc](https://www.geeksforgeeks.org/protected-variable-in-python/)
Tu pourrais alors renommer ce membre de classe "outil" et lui ajouter un `_` en préfixe.